astroid/brain/brain_typing.py

```diff
@@ -344,6 +346,34 @@ def infer_tuple_alias(
     return iter([class_def])
 
 
+def _looks_like_typing_cast(node: Call) -> bool:
+    return isinstance(node, Call) and (
+        isinstance(node.func, Name)
+        and node.func.name == "cast"
+        or isinstance(node.func, Attribute)
+        and node.func.attrname == "cast"
+    )
+
+
+def infer_typing_cast(
+    node: Call, ctx: context.InferenceContext = None
+) -> typing.Iterator[Instance]:
```

Comment:
```suggestion
) -> typing.Union[typing.Iterator[NodeNG], typing.Iterator[Instance]]:
```
Not completely sure about this as `infer` isn't yet typed.